### PR TITLE
New runtime `spec_version` format + backport of the bump to 1.4.0

### PR DESCRIPTION
**Pull request by chevdor**
## Overview

This PR aligns the `spec_version` formatting to the [recent changes](https://github.com/polkadot-fellows/runtimes/pull/26/files#diff-efa4caeb17487ecb13d8f5eb7863c3241d84afa2e73fbf25909a2ca89df0f362R142) made for the Polkadot/Kusama runtimes. 

It also backports the latest version `v1.4.0` bumps as `1_004_000`.

## Details

During the switch from `v0.9` to `v1.x`, the format of the `spec_version` was modified from: `(M)m_ppp` for a runtime considered on version `M.m.pp`. For instance `0.9.42` had a `spec_version` of `9420`.

With the transition to `v1.x`, the format was changed to a bigger number (still `u32`) formatted as `MM_mm_ppp` where `1.2.3` would be stored as `01_02_003`.

This PR aligns the format with that has been introduced in the fellowship repo: `MMM_mmm_ppp`.

### CODE REVIEW

**Review by joepetrowski — APPROVED** [review]

**Review by lexnv — APPROVED** [review]

**Review by coderobe — APPROVED** [review]

File: substrate/bin/node-template/runtime/src/lib.rs [hunk 1]
```diff
@@ -105,7 +105,7 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
 	//   `spec_version`, and `authoring_version` are the same between Wasm and native.
 	// This value is set to 100 to notify Polkadot-JS App (https://polkadot.js.org/apps) to use
 	//   the compatible custom types.
-	spec_version: 100,
+	spec_version: 0_000_100,
```
Comment (bkchr) [inline substrate/bin/node-template/runtime/src/lib.rs:108]:
```suggestion
	spec_version: 1,
```

File: cumulus/parachain-template/runtime/src/lib.rs [hunk 2]
```diff
@@ -184,7 +184,7 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
 	spec_name: create_runtime_str!("template-parachain"),
 	impl_name: create_runtime_str!("template-parachain"),
 	authoring_version: 1,
-	spec_version: 1,
+	spec_version: 0_000_001,
```
Comment (bkchr) [inline cumulus/parachain-template/runtime/src/lib.rs:187]:
```suggestion
	spec_version: 1,
```

File: polkadot/runtime/test-runtime/src/lib.rs [hunk 3]
```diff
@@ -107,7 +107,7 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
 	spec_name: create_runtime_str!("polkadot-test-runtime"),
 	impl_name: create_runtime_str!("parity-polkadot-test-runtime"),
 	authoring_version: 2,
-	spec_version: 1056,
+	spec_version: 1_004_000,
```
Comment (bkchr) [inline polkadot/runtime/test-runtime/src/lib.rs:110]:
```suggestion
	spec_version: 1056,
```

File: cumulus/parachains/runtimes/testing/penpal/src/lib.rs [hunk 4]
```diff
@@ -230,7 +230,7 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
 	spec_name: create_runtime_str!("penpal-parachain"),
 	impl_name: create_runtime_str!("penpal-parachain"),
 	authoring_version: 1,
-	spec_version: 10000,
+	spec_version: 1_004_000,
```
Comment (bkchr) [inline cumulus/parachains/runtimes/testing/penpal/src/lib.rs:233]:
```suggestion
	spec_version: 1,
```

File: cumulus/parachains/runtimes/starters/seedling/src/lib.rs [hunk 5]
```diff
@@ -75,7 +75,7 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
 	spec_name: create_runtime_str!("seedling"),
 	impl_name: create_runtime_str!("seedling"),
 	authoring_version: 1,
-	spec_version: 10000,
+	spec_version: 1_004_000,
```
Comment (bkchr) [inline cumulus/parachains/runtimes/starters/seedling/src/lib.rs:78]:
```suggestion
	spec_version: 1,
```

File: substrate/primitives/version/src/lib.rs [hunk 6]
```diff
@@ -182,6 +182,9 @@ pub struct RuntimeVersion {
 	/// between Wasm and native.
 	///
 	/// This number should never decrease.
+	///
+	/// This number is formatted as `u32` in a semver looking manner. For instance, `1.2.3` would
+	/// be encoded as `001_002_003`.
```
Comment (bkchr) [inline substrate/primitives/version/src/lib.rs:187]:
```suggestion
```
Please don't change docs based on some higher level concept that are not related to this number.

**Review by bkchr — APPROVED** [review]

File: substrate/bin/node-template/runtime/src/lib.rs [hunk 7]
```diff
@@ -105,7 +105,7 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
 	//   `spec_version`, and `authoring_version` are the same between Wasm and native.
 	// This value is set to 100 to notify Polkadot-JS App (https://polkadot.js.org/apps) to use
 	//   the compatible custom types.
-	spec_version: 100,
+	spec_version: 1,
```
Comment (ordian) [inline substrate/bin/node-template/runtime/src/lib.rs:108]: the comment says it should be 100. will this also be an issue for other runtime that you set version to 1?

Comment (bkchr) [reply substrate/bin/node-template/runtime/src/lib.rs:108]: FML

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4470116

File: substrate/bin/node-template/runtime/src/lib.rs (hunk 7 above)
Comment (bkchr) [inline substrate/bin/node-template/runtime/src/lib.rs:108]:
```suggestion
	spec_version: 100,
```

File: cumulus/parachains/runtimes/starters/seedling/src/lib.rs [hunk 8]
```diff
@@ -75,7 +75,7 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
 	spec_name: create_runtime_str!("seedling"),
 	impl_name: create_runtime_str!("seedling"),
 	authoring_version: 1,
-	spec_version: 10000,
+	spec_version: 1,
```
Comment (ordian) [inline cumulus/parachains/runtimes/starters/seedling/src/lib.rs:78]: you're saying this won't be a problem here?

File: cumulus/parachains/runtimes/testing/penpal/src/lib.rs [hunk 9]
```diff
@@ -230,7 +230,7 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
 	spec_name: create_runtime_str!("penpal-parachain"),
 	impl_name: create_runtime_str!("penpal-parachain"),
 	authoring_version: 1,
-	spec_version: 10000,
+	spec_version: 1,
```
Comment (ordian) [inline cumulus/parachains/runtimes/testing/penpal/src/lib.rs:233]: and here?

**Review by ordian — APPROVED** [review]

File: cumulus/parachains/runtimes/testing/penpal/src/lib.rs (hunk 9 above)
Comment (bkchr) [reply cumulus/parachains/runtimes/testing/penpal/src/lib.rs:233]: This is a test runtime, that should never require a spec version bump.

File: cumulus/parachains/runtimes/starters/seedling/src/lib.rs (hunk 8 above)
Comment (bkchr) [reply cumulus/parachains/runtimes/starters/seedling/src/lib.rs:78]: This is more like a template runtime.
